pshtt/pshtt.py

```diff
@@ -654,88 +685,51 @@ def https_check(endpoint):
         logging.exception("{}: Unknown exception examining trust.".format(endpoint.url))
         utils.debug("{}: Unknown exception examining trust: {}".format(endpoint.url, err))
 
-    try:
-        cert_response = cert_plugin_result.as_text()
-    except AttributeError:
-        logging.exception("{}: Known error in sslyze 1.X with EC public keys. See https://github.com/nabla-c0d3/sslyze/issues/215".format(endpoint.url))
-        return None
-    except Exception as err:
-        endpoint.unknown_error = True
-        logging.exception("{}: Unknown exception in cert plugin.".format(endpoint.url))
-        utils.debug("{}: {}".format(endpoint.url, err))
-        return
-
-    # Debugging
-    # for msg in cert_response:
-    #     print(msg)
-
     # Default endpoint assessments to False until proven True.
     endpoint.https_expired_cert = False
     endpoint.https_self_signed_cert = False
     endpoint.https_bad_chain = False
     endpoint.https_bad_hostname = False
 
-    # STORE will be either "Mozilla" or "Custom"
-    # depending on what the user chose.
-
-    # A certificate can have multiple issues.
-    for msg in cert_response:
-
-        # Check for missing SAN.
-        if (
-            (("DNS Subject Alternative Names") in msg) and
-            (("[]") in msg)
-        ):
-            endpoint.https_bad_hostname = True
-
-        # Check for certificate expiration.
-        if (
-            (STORE in msg) and
-            (("FAILED") in msg) and
-            (("certificate has expired") in msg)
-        ):
-            endpoint.https_expired_cert = True
+    cert_chain = cert_plugin_result.certificate_deployments[0].received_certificate_chain
```

Comment:
Using `.received_certificate_chain` vs using `.path_validation_results, filtering on `.trust_store.name == STORE` for each element of that list, and then using `.verified_certificate_chain` ignores the usage of `STORE` before this PR in these checks that follow. We care about the result from the `Mozilla` trust store, or a `Custom` store that is provided.

@jsf9k I would really like your thoughts here.

Reply:
I think going `.path_validation_results` sounds like it will be more consistent with functionalities before this PR. So maybe I will just implement that flow and check with `.trust_store.name` to take the STORE into consideration. cc @mcdonnnj @jsf9k

Reply:
@mcdonnnj like in the other comment, should we consider all certificate deployments here as well?

Reply:
@SaptakS - yes, let's keep it consistent with the functionality before the PR.  Let's also consider all certificate deployments.